shell/platform/windows/platform_handler_win32_unittests.cc

```diff
@@ -0,0 +1,239 @@
+
+  auto args = std::make_unique<rapidjson::Document>(rapidjson::kStringType);
+  auto& allocator = args->GetAllocator();
+  args->SetString(kTextPlainFormat);
+  auto encoded = JsonMethodCodec::GetInstance().EncodeMethodCall(
+      MethodCall<rapidjson::Document>(kHasStringsClipboardMethod,
+                                      std::move(args)));
+
+  MockMethodResult result;
+  rapidjson::Document document;
+  document.SetObject();
+  rapidjson::Document::AllocatorType& document_allocator =
+      document.GetAllocator();
+  document.AddMember(rapidjson::Value(kValueKey, document_allocator),
+                     rapidjson::Value(false), document_allocator);
+
+  EXPECT_CALL(result, SuccessInternal(_))
+      .WillOnce([](const rapidjson::Document* document) {
+        ASSERT_TRUE((*document)[kValueKey].GetBool());
+      });
+  EXPECT_TRUE(messenger.SimulateEngineMessage(
+      kChannelName, encoded->data(), encoded->size(),
+      [&](const uint8_t* reply, size_t reply_size) {
+        JsonMethodCodec::GetInstance().DecodeAndProcessResponseEnvelope(
+            reply, reply_size, &result);
+      }));
+}
+
+TEST(PlatformHandlerWin32, HasStringsSuccessWithoutStrings) {
+  TestBinaryMessenger messenger;
+  FlutterWindowsView view(
+      std::make_unique<::testing::NiceMock<MockWindowBindingHandler>>());
+  // HasStrings will succeed and return false.
+  PlatformHandlerWin32 platform_handler(
+      &messenger, &view,
+      std::make_unique<TestScopedClipboard>(kErrorSuccess, false));
+
+  auto args = std::make_unique<rapidjson::Document>(rapidjson::kStringType);
+  auto& allocator = args->GetAllocator();
+  args->SetString(kTextPlainFormat);
+  auto encoded = JsonMethodCodec::GetInstance().EncodeMethodCall(
+      MethodCall<rapidjson::Document>(kHasStringsClipboardMethod,
+                                      std::move(args)));
+
+  MockMethodResult result;
+  rapidjson::Document document;
+  document.SetObject();
+  rapidjson::Document::AllocatorType& document_allocator =
+      document.GetAllocator();
+  document.AddMember(rapidjson::Value(kValueKey, document_allocator),
+                     rapidjson::Value(false), document_allocator);
+
+  EXPECT_CALL(result, SuccessInternal(_))
+      .WillOnce([](const rapidjson::Document* document) {
+        ASSERT_FALSE((*document)[kValueKey].GetBool());
+      });
+  EXPECT_TRUE(messenger.SimulateEngineMessage(
+      kChannelName, encoded->data(), encoded->size(),
+      [&](const uint8_t* reply, size_t reply_size) {
+        JsonMethodCodec::GetInstance().DecodeAndProcessResponseEnvelope(
+            reply, reply_size, &result);
+      }));
+}
+
+TEST(PlatformHandlerWin32, HasStringsError) {
+  TestBinaryMessenger messenger;
+  FlutterWindowsView view(
+      std::make_unique<::testing::NiceMock<MockWindowBindingHandler>>());
+  // HasStrings will fail.
+  PlatformHandlerWin32 platform_handler(
+      &messenger, &view, std::make_unique<TestScopedClipboard>(1, true));
```

Comment:
To make this self-documenting, I would make a local `const int arbitraryErrorCode = 1;` and pass that in.